src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -310,6 +321,41 @@
     /** Actions to add to the {@link FlowStartNode}. */
     transient final List<Action> flowStartNodeActions = new ArrayList<Action>();
 
+    enum TimingKind {
+        /**
+         * Parsing Groovy sources; includes {@link #load}.
+         * @see CpsGroovyShell#parse(GroovyCodeSource)
+         */
+        parse,
+        /**
+         * Loading classes needed during {@link #parse}.
+         * @see ClassLoader#loadClass(String, boolean)
+         * @see ClassLoader#getResource
+         */
+        load,
+        /**
+         * Running inside {@link CpsVmExecutorService}, which includes many other things.
+         */
+        run,
+        /**
+         * Saving the program state.
+         * @see CpsThreadGroup#saveProgram(File)
+         */
+        save,
+        /**
+         * Loading or saving flow nodes.
+         * @see FlowNodeStorage
+         */
+        flow
```

Comment:
🐛  better to call it flowNodeIO... again because there are parts of the logic that interact with the flow (iteration, in-memory work) which are separate from I/O routines.